src/rgw/rgw_rest_s3.cc

```diff
@@ -3574,18 +3574,25 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
 RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
 {
   // Non-website mode
-  int list_type = 1;
+  int list_type;
   s->info.args.get_int("list-type", &list_type, 1);
 
    // Non-website mode    // Non-website mode
   if (get_data) {   
     if (list_type == 1) {
-       return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
+      return new RGWListBucket_ObjStore_S3;
+    }
+    else if (list_type == 2) {
```

Comment:
please move the `else if` up into the previous line.  also i'd recommend use `switch-case`.